### PR TITLE
basic axios utility functions

### PR DESCRIPTION
**Pull request by AedonGrunGott**
I suggest we all use this pattern for requets. Import the utility file, and handle data/errors inside your own component.

Less code to maintain, and any errors that appear will clearly show inside the component that needs attention.

### CODE REVIEW

**Comment by AedonGrunGott** [comment]
first pass, does not yet support POST as I can't get that to work in Postman

File: client/source/utilities/axiosRequests.js [hunk 1]
```diff
@@ -1,18 +1,52 @@
...
+ * Fire an Axios Get Request.
+ *
+ * It takes a string value to append to the endpoint,
+ * which should match the Atelier API endpoint pattern.
+ * @param {string} endpoint - Endpoint to be used
+ * @returns an Axios Request Promise
+ * */
 const get = function getByEndpoint(endpoint) {
-  return axios.get(`${authServerURL}/${endpoint}`);
+  return axios.get(`${authServerURL}/api/${endpoint}`);
```
Comment (aimeekang) [inline client/source/utilities/axiosRequests.js]: A suggestion to leave the slash after api out from the axiosRequests.js file? I'm thinking about when any one of us makes a request on our module, I'd like to use /endpoint vs. just endpoint, as I use the variable "reviews", which also happens to be the endpoint I'm using. This way, the /reviews can help distinguish.

File: client/source/utilities/TestAPI.jsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+import * as requests from './axiosRequests';
+
+function testGetRequest() {
+  requests.get('products')
```
Comment (aimeekang) [inline client/source/utilities/TestAPI.jsx]: Update to .get('/products') based on proposed change.

File: client/source/utilities/TestAPI.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    .then((response) => console.info(response.status, response.data))
+    .catch((error) => error);
+}
+
+function testPutRequest() {
+  requests.put('reviews/1278333/helpful', { review_id: 0 })
```
Comment (aimeekang) [inline client/source/utilities/TestAPI.jsx]: Update to .put('/reviews/1278333/helpful') based on proposed change.

File: client/source/utilities/TestAPI.jsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    .then((response) => console.info(response.status, response.data))
```
Comment (aimeekang) [inline client/source/utilities/TestAPI.jsx:35]: Thank you so much for including these tests! Super helpful examples. 

No action, just an observation. I saw this logged `204 ''` in my browser console. I think per Learn, we may want to use Status: 204 NO CONTENT. We can probably each handle this on client-side ourselves when we do a .then and .catch.

File: client/source/utilities/axiosRequests.js (hunk 1 above)
Comment (AedonGrunGott) [reply client/source/utilities/axiosRequests.js]: Easy fix, sure!

File: server/controllers/index.js [hunk 5]
```diff
@@ -27,7 +27,26 @@ function parseGet(request, response) {
...
+    headers: {
+      Authorization: process.env.API_TOKEN,
+    },
+  };
+
+  axios.put(newURL, request.body, config)
+    .then((apiRes) => {
+      console.info('request ended in', apiRes.status);
+      response.status(apiRes.status);
+      response.send(apiRes.data);
```
Comment (aimeekang) [inline server/controllers/index.js:61]: Could we adjust this to the below (per Learn): 

Status: 204 NO CONTENT

File: client/source/utilities/TestAPI.jsx (hunk 4 above)
Comment (AedonGrunGott) [reply client/source/utilities/TestAPI.jsx:35]: I'm not sure what you mean by this. The status code `204` can only mean `no content`, they are one and the same.

**Review by aimeekang — CHANGES_REQUESTED** [review]
Thank you for setting this up. Love the idea of centralized axios requests. Helps for better debugging, and creates consistency on how we make these requests. Few minor comments and suggestions.

File: server/controllers/index.js (hunk 5 above)
Comment (AedonGrunGott) [reply server/controllers/index.js:61]: ... as above.

File: client/source/utilities/TestAPI.jsx (hunk 4 above)
Comment (aimeekang) [reply client/source/utilities/TestAPI.jsx:35]: I was suggesting maybe we actually log the string 'NO CONTENT', but this is fine.

File: server/controllers/index.js (hunk 5 above)
Comment (aimeekang) [reply server/controllers/index.js:61]: See response above.

File: client/source/utilities/axiosRequests.js [hunk 6]
```diff
@@ -1,18 +1,52 @@
...
+/**
+ * Fire an Axios Post Request.
+ *
+ * It takes a data json, and a string value to append to the endpoint,
+ * which should match the Atelier API endpoint pattern.
+ * @param {string} endpoint - Endpoint to be used
+ * @param {json} json - Data to be posted
+ * @returns an Axios Request Promise
+ * */
+const post = function postBygetByEndpoint(endpoint, json) {
```
Comment (aimeekang) [inline client/source/utilities/axiosRequests.js:27]: I saw one of your commit comments on not including a POST. Do you need a test example? I got one to work with posting a review. Just lmk!

Comment (AedonGrunGott) [reply client/source/utilities/axiosRequests.js:27]: Yes please! It's been bothering me that I can't get the POST changes to be present in my later GET requests. I must have malformed data somehow.

**Comment by AedonGrunGott** [comment]
@aimeekang requests committed, please review!

**Review by aimeekang — APPROVED** [review]
Thanks for the changes and adding in a POST test! Looks good.

**Pull request merged**